User page

liornoy opened the pull request:
A user page that shows all the users,
and an individual user page for each user, that shows:
1. username
2. join date
3. num of answers - and the answers
4. num of questions - and the questions
5. gender
added also links to everywhere a user name is shown
You can also get to your own profile page by clicking you username in the right side of the navbar

fix #147

avivz450 commented on home/templates/home/user_page.html:
```diff
@@ -0,0 +1,26 @@
+<link rel="stylesheet" type="text/css" href="{% static 'home/CSS/user_page.css' %}">
+<body class="main-bg">
+<main role="main" class="container">
+ <h1 class="title">{{ profile }} Profile page</h1>
```
If the profile name is "Lior", we get from this line: "Lior Profile page".
The reason of that is because __str__ is returning `'{self.user.username}'.format(self=self)`.
The problem is that if `__str__` will be changed in the future, we will get a different output here.
Therefore, I think it should be {{ profile.user.username }} instead of {{ profile }}.

liornoy replied:
great comment

avivz450 commented on home/templates/home/user_page.html:
```diff
@@ -0,0 +1,26 @@
+ <h1 class="title">{{ profile }} Profile page</h1>
+<div class="row">
+  <div class="column left-col border border-dark" >
+   <h2>User Questions</h2>
+   {% for question in user_questions %}
+		<a class="questions" href="{% url 'question-detail' question.pk  %}">{{question}}</a>
+		<br/>
+	{% endfor %}
+  </div>
+  <div class="column right-col border border-dark">
+  <h2>User Answers</h2>
+  {% for answer in user_answers %}
+		<a class="answers" href="{% url 'question-detail' answer.question.pk  %}">Answered {{answer.question}}</a>
+		<br/>
+	{% endfor %}
```
Maybe we should put more data on the profile's page besides the user's answers and questions.
I thought about adding as well: num of answers, num of questions, registration date, and email.
What do you think?

liornoy replied:
I'm down to add num of.. and registration date, email is something that should be kept privately IMO, and maybe I can add the gender with an adjacent icon if we already have this info.

avivz450 replied:
Ok, it is possible at right now not to display this data and in the future maybe we will add an option to ask the user which data he prefers to be displayed and which not.

avivz450 commented on home/models.py:
```diff
@@ -137,6 +141,10 @@ def get_question_title(self):
 
+    @classmethod
+    def get_all_user_questions(cls, profile):
+        return cls.objects. all().filter(profile=profile)
```
there is a space here

liornoy replied:
thanks fixed

avivz450 reviewed (approved):
Please edit the description of the PR with the new data that is displayed on the user page before merging

liornoy commented:
@avivz450 I added a link from the username in the top right in the navbar to the profile page. and added reverse function to the tests. is it cool with you?

avivz450 commented:
> @avivz450 I added a link from the username in the top right in the navbar to the profile page. and added reverse function to the tests. is it cool with you?

yep
Please try next time to make the PRs smaller so it will be easier to review (and with fewer commits).
